data-prepper-plugins/kinesis-source/src/main/java/org/opensearch/dataprepper/plugins/kinesis/extension/KinesisLeaseConfigSupplier.java

```diff
@@ -0,0 +1,10 @@
+package org.opensearch.dataprepper.plugins.kinesis.extension;
+
+import java.util.Optional;
+
+public interface KinesisLeaseConfigSupplier {
+
+    default Optional<KinesisLeaseConfig> getKinesisExtensionLeaseConfig() {
```

Comment:
Why did you make a default implementation? I think just require an implementation here.